pkg/controller/direct/essentialcontacts/contact_controller.go

```diff
@@ -0,0 +1,292 @@
+	}
+
+	a.actual = actual
+	return true, nil
+}
+
+// Create creates the resource in GCP based on `spec` and update the Config Connector object `status` based on the GCP response.
+func (a *contactAdapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("creating essential contact", "name", a.id)
+	mapCtx := &direct.MapContext{}
+
+	desired := a.desired.DeepCopy()
+	resource := EssentialContactsContactSpec_ToProto(mapCtx, &desired.Spec)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	req := &pb.CreateContactRequest{
+		Parent:  a.id.Parent().String(),
+		Contact: resource,
+	}
+	created, err := a.gcpClient.CreateContact(ctx, req)
+	if err != nil {
+		return fmt.Errorf("creating essential contact %s: %w", a.id, err)
+	}
+	log.V(2).Info("successfully created essential contact", "name", a.id)
+
+	status := &krm.EssentialContactsContactStatus{}
+	status.ObservedState = EssentialContactsContactObservedState_FromProto(mapCtx, created)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	status.ExternalRef = direct.LazyPtr(created.Name)
+	return createOp.UpdateStatus(ctx, status, nil)
+}
+
+// Update updates the resource in GCP based on `spec` and update the Config Connector object `status` based on the GCP response.
+func (a *contactAdapter) Update(ctx context.Context, updateOp *directbase.UpdateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("updating essential contact", "name", a.id)
+	mapCtx := &direct.MapContext{}
+
+	desired := a.desired.DeepCopy()
+	resource := EssentialContactsContactSpec_ToProto(mapCtx, &desired.Spec)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	resource.Name = a.id.String() // Set name for update request
+
+	paths := []string{}
+	if !reflect.DeepEqual(resource.NotificationCategorySubscriptions, a.actual.NotificationCategorySubscriptions) {
+		paths = append(paths, "notification_category_subscriptions")
+	}
+	if !reflect.DeepEqual(resource.LanguageTag, a.actual.LanguageTag) {
+		paths = append(paths, "language_tag")
+	}
+
+	var updated *pb.Contact
+	if len(paths) == 0 {
+		log.V(2).Info("no field needs update", "name", a.id)
+		updated = a.actual
+	} else {
+		req := &pb.UpdateContactRequest{
+			Contact:    resource,
+			UpdateMask: &fieldmaskpb.FieldMask{Paths: paths},
+		}
+		var err error
+		updated, err = a.gcpClient.UpdateContact(ctx, req)
+		if err != nil {
+			return fmt.Errorf("updating essential contact %s: %w", a.id, err)
+		}
+		log.V(2).Info("successfully updated essential contact", "name", a.id)
+	}
+
+	status := &krm.EssentialContactsContactStatus{}
+	status.ObservedState = EssentialContactsContactObservedState_FromProto(mapCtx, updated)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	status.ExternalRef = direct.LazyPtr(updated.Name)
+
+	// Handle potential status-only updates if KCC resource was acquired.
+	if len(paths) == 0 && a.desired.Status.ExternalRef == nil {
```

Comment:
Not sure I follow this. `a.desired.Status` is already nil?

Reply:
you are correct. removed.,